zerver/webhooks/gogs/view.py

```diff
@@ -63,18 +63,22 @@ def format_pull_request_event(payload):
 @has_request_variables
 def api_gogs_webhook(request, user_profile, client,
                      payload=REQ(argument_type='body'),
-                     stream=REQ(default='commits')):
-    # type: (HttpRequest, UserProfile, Client, Dict[str, Any], Text) -> HttpResponse
+                     stream=REQ(default='commits'),
+                     branches=REQ(default=None)):
+    # type: (HttpRequest, UserProfile, Client, Dict[str, Any], Text, Optional[Text]) -> HttpResponse
 
     repo = payload['repository']['name']
     event = request.META['HTTP_X_GOGS_EVENT']
 
     try:
         if event == 'push':
+            branch = payload['ref'].replace('refs/heads/', '')
+            if branches is not None and branch not in branches.split(','):
```

Comment:
instead of spliting string I'd using find:
`branches.find(branch) >= 0`
`find` returns index that the branch started to be so it might be 0 as well.